### PR TITLE
Add types for three-gif-loader

### PR DESCRIPTION
**Pull request by lairjr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/three-gif-loader/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    load(
+        url: string,
+        onLoad: (reader: GifReader) => void,
+        onProgress?: (request: ProgressEvent) => void,
+        onError?: (event: ErrorEvent) => void,
+    ): GifTexture;
+    setPath(value: string): void;
+}
+
+export class GifTexture extends CanvasTexture {
```
Comment (andrewbranch) [inline types/three-gif-loader/index.d.ts:20]: I don’t see this exported from index.js https://unpkg.com/browse/three-gif-loader@1.1.0/index.js

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/three-gif-loader/index.d.ts (hunk 1 above)
Comment (lairjr) [reply types/three-gif-loader/index.d.ts]: Done. Thank you!

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/three-gif-loader/src/gif-texture.d.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { CanvasTexture } from 'three';
+import { GifReader } from 'omggif';
+
+export default class GifTexture extends CanvasTexture {
```
[the remark this answers is not in the document]
Comment (lairjr) [reply types/three-gif-loader/src/gif-texture.d.ts]: Thank you.

**Review by andrewbranch — APPROVED** [review]
Thanks!

**Comment by lairjr** [comment]
Ready to merge

**Pull request merged**